lib/rom/sql/function.rb

```diff
@@ -26,6 +26,21 @@ def is(other)
         ::Sequel::SQL::BooleanExpression.new(:'=', func, other)
       end
 
+      # Convert an expression result to another data type
+      #
+      # @example
+      #   users.select { bool::cast(json_data.get_text('activated'), :boolean).as(:activated) }
```

Comment:
you're absolutely right, but this will require a DB-specific mapping between data types (e.g. oracle uses `varchar2` instead of `varchar`), that's why I decided to make it explicit for now, I'll open a help-requested issue for this improvement :)